features/filesystem/littlefs/LittleFileSystem.h

```diff
@@ -0,0 +1,283 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2012 ARM Limited
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#ifndef MBED_LFSFILESYSTEM_H
+#define MBED_LFSFILESYSTEM_H
+
+#include "FileSystem.h"
+#include "BlockDevice.h"
+#include "PlatformMutex.h"
+extern "C" {
+#include "lfs.h"
+}
+
+using namespace mbed;
```

Comment:
Please not in headers: it polute all source files including LittleFS and avoiding [using statements](https://github.com/ARMmbed/mbed-os/blob/master/mbed.h#L99-L100) in mbed.h. There's no point in having namespace if those are automatically brought automatically and entirely in the global namespace when you include files.

Reply:
Ah good catch, put it in during prototyping and forgot to take it out 👍 

will update